apoc.atomic.add and related procedures  do not work with Integer properties

## Expected Behavior (Mandatory)

When a Node has a property with an Integer value (which is allowed by Neo4J PropertyContainer.setProperty()), the apoc.atomic.add() procedure should work.

## Actual Behavior (Mandatory)

An IllegalArgumentException is thrown :

<pre>
2019-04-09 03:22:01.901+0000 ERROR org.neo4j.graphdb.QueryExecutionException: Failed to invoke procedure `apoc.atomic.add`: Caused by: java.lang.IllegalArgumentException: [null] is not a supported property value
	at [...]
Caused by: java.lang.IllegalArgumentException: [null] is not a supported property value
	at org.neo4j.values.storable.Values.unsafeOf(Values.java:594)
	at org.neo4j.values.storable.Values.of(Values.java:516)
	at org.neo4j.kernel.impl.core.NodeProxy.setProperty(NodeProxy.java:242)
	at apoc.atomic.Atomic.lambda$add$0(Atomic.java:47
	at apoc.atomic.Atomic.lambda$retry$6(Atomic.java:226)
	at org.neo4j.helpers.TransactionTemplate.lambda$execute$2(TransactionTemplate.java:179)
	at org.neo4j.helpers.TransactionTemplate.execute(TransactionTemplate.java:201)
	... 82 more
</pre>

## How to Reproduce the Problem

Create a node with an int property (use node.setProperty with an int argument), and try and invoke apoc.atomic.add on the node for that property.

### Steps (Mandatory)

  1. Create Node
  2. Set a property to an int value
  3. Commit
  3. Retrieve the node and invoke apoc.atomic.add on it

### Root Cause

The AtomicUtils class in apoc.atomic.util returns <code>null</code> if the type of the property is not one of the 4 types it expects (Long, Double, Short, Byte). Integer, Float and Short should be supported too since they can all occur as types of a Property value in a PropertyContainer.

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: OSX
  - Neo4j: 3.5.3
  - Neo4j-Apoc: 3.5.0.2

--- a/core/src/main/java/apoc/atomic/util/AtomicUtils.java
+++ b/core/src/main/java/apoc/atomic/util/AtomicUtils.java
@@ -1,6 +1,5 @@
 package apoc.atomic.util;
 
-
 /**
  * @author AgileLARUS
  *

--- a/core/src/main/java/apoc/atomic/util/AtomicUtils.java
+++ b/core/src/main/java/apoc/atomic/util/AtomicUtils.java
@@ -1,5 +1,6 @@
 package apoc.atomic.util;
 
+
 /**
  * @author AgileLARUS
  *
@@ -10,8 +11,12 @@ public class AtomicUtils {
 	public static Number sum(Number oldValue, Number number){
 		if(oldValue instanceof Long)
 			return oldValue.longValue() + number.longValue();
+		if(oldValue instanceof Integer)
+			return oldValue.intValue() + number.intValue();
 		if(oldValue instanceof Double)
 			return oldValue.doubleValue() + number.doubleValue();
+		if(oldValue instanceof Float)
+			return oldValue.floatValue() + number.floatValue();
 		if(oldValue instanceof Short)
 			return oldValue.shortValue() + number.shortValue();
 		if(oldValue instanceof Byte)
@@ -22,8 +27,12 @@ public class AtomicUtils {
 	public static Number sub(Number oldValue, Number number){
 		if(oldValue instanceof Long)
 			return oldValue.longValue() - number.longValue();
+		if(oldValue instanceof Integer)
+			return oldValue.intValue() - number.intValue();
 		if(oldValue instanceof Double)
 			return oldValue.doubleValue() - number.doubleValue();
+		if(oldValue instanceof Float)
+			return oldValue.floatValue() - number.floatValue();
 		if(oldValue instanceof Short)
 			return oldValue.shortValue() - number.shortValue();
 		if(oldValue instanceof Byte)

--- a/core/src/test/java/apoc/atomic/AtomicTest.java
+++ b/core/src/test/java/apoc/atomic/AtomicTest.java
@@ -35,6 +35,40 @@ public class AtomicTest {
 	}
 
 	@Test
+	public void testAddAndSubIngeger(){
+		db.executeTransactionally("CREATE (p:Person {name:'Tom'})");
+		try(Transaction tx = db.beginTx()){
+			final Node node = tx.getAllNodes().stream().findFirst().orElse(null);
+			node.setProperty("age", 1);
+			tx.commit();
+		}
+		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",10), (r) -> {});
+		int age = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
+		assertEquals(11, age);
+		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",5), (r) -> {});
+		int ageSub = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
+		assertEquals(6, ageSub);
+	}
+
+	@Test
+	public void testAddAndSubFloat() {
+		db.executeTransactionally("CREATE (p:Person {name:'Tom'})");
+		try(Transaction tx = db.beginTx()){
+			final Node node = tx.getAllNodes().stream().findFirst().orElse(null);
+			node.setProperty("age", 2.4F);
+			tx.commit();
+		}
+
+		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",1.5F), (r) -> {});
+		float age = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
+		assertEquals(3.9F, age, 0);
+		
+		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",0.5F), (r) -> {});
+		float ageSub = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
+		assertEquals(3.4F, ageSub, 0);
+	}
+
+	@Test
 	public void testAddLong(){
 		db.executeTransactionally("CREATE (p:Person {name:'Tom',age: 40}) CREATE (c:Person {name:'John',age: 40}) CREATE (a:Person {name:'Anne',age: 22})");
 		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",10), (r) -> {});

--- a/core/src/test/java/apoc/atomic/AtomicTest.java
+++ b/core/src/test/java/apoc/atomic/AtomicTest.java
@@ -35,17 +35,19 @@ public class AtomicTest {
 	}
 
 	@Test
-	public void testAddAndSubIngeger(){
+	public void testAddAndSubInteger(){
 		db.executeTransactionally("CREATE (p:Person {name:'Tom'})");
-		try(Transaction tx = db.beginTx()){
+		try(Transaction tx = db.beginTx()) {
 			final Node node = tx.getAllNodes().stream().findFirst().orElse(null);
 			node.setProperty("age", 1);
 			tx.commit();
 		}
-		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",10), (r) -> {});
+		db.executeTransactionally("MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",
+				map("property","age","value",10));
 		int age = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
 		assertEquals(11, age);
-		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",5), (r) -> {});
+		db.executeTransactionally("MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",
+				map("property","age","value",5));
 		int ageSub = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
 		assertEquals(6, ageSub);
 	}
@@ -53,17 +55,19 @@ public class AtomicTest {
 	@Test
 	public void testAddAndSubFloat() {
 		db.executeTransactionally("CREATE (p:Person {name:'Tom'})");
-		try(Transaction tx = db.beginTx()){
+		try(Transaction tx = db.beginTx()) {
 			final Node node = tx.getAllNodes().stream().findFirst().orElse(null);
 			node.setProperty("age", 2.4F);
 			tx.commit();
 		}
 
-		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",1.5F), (r) -> {});
+		db.executeTransactionally("MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.add(n,$property,$value) YIELD container RETURN count(*)",
+				map("property","age","value",1.5F));
 		float age = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
 		assertEquals(3.9F, age, 0);
-		
-		testCall(db, "MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",map("property","age","value",0.5F), (r) -> {});
+
+		db.executeTransactionally("MATCH (n:Person {name:'Tom'}) CALL apoc.atomic.subtract(n,$property,$value) YIELD container RETURN count(*)",
+				map("property","age","value",0.5F));
 		float ageSub = TestUtil.singleResultFirstColumn(db, "MATCH (n:Person {name:'Tom'}) RETURN n.age as age");
 		assertEquals(3.4F, ageSub, 0);
 	}

